src/java/com/zimbra/cs/nginx/NginxLookupExtension.java

```diff
@@ -1046,7 +1046,9 @@ private void search(NginxLookupRequest req) throws NginxLookupException {
                         if (doDnsLookup) {
                             upstreamHost = this.getIPByIPMode(upstreamHost).getHostAddress();
                         }
-                        String upstreamPort = req.proto.equals(IMAP) ? imapServer.getImapBindPortAsString() : imapServer.getImapSSLBindPortAsString();
+                        String upstreamPort = req.proto.equals(IMAP) ?
```

Comment:
As far as I can see, this code will only be executed when zimbraReverseProxyUpstreamImapServers is set and this setting should not be pointing to IMAP running inside mailboxd.